app/src/main/kotlin/com/flxrs/dankchat/channels/ChannelsAdapter.kt

```diff
@@ -22,7 +21,16 @@ class ChannelsAdapter(private val onEditChannel: (String, String?) -> Unit) : Li
         val dankChatPreferences = DankChatPreferenceStore(holder.itemView.context)
         val channel = getItem(position)
         val renamedChannel = dankChatPreferences.getRenamedChannel(channel)
-        channelText.text = renamedChannel ?: channel
+        when (renamedChannel) {
```

Comment:
You don't have to add a new textview for the italic text. Spans are a perfect fit here. 
Something like this should work:
```kotlin
channelText.text = buildSpannedString {
    append(renamedChannel ?: channel)
    renamedChannel?.let {
        italic { it }
    }
}
```

Reply:
This works for the italic text, but I couldn't find a function to modify the alpha of the text this way. I guess I could use ``channelText.background.alpha`` but that would set the alpha for the entire text and not just for the original channel text. This is what it looks like without the transparent effect: 

![](https://cdn.discordapp.com/attachments/503657071196110865/889232384443158528/1632079916367.jpg)

Reply:
That should be doable with spans as well. Something like this:
```kotlin
channelText.text = buildSpannedString {
    append(renamedChannel ?: channel)
    renamedChannel?.let {
        val channelColor = ColorUtils.setAlphaComponent(channelText.currentTextColor, 128)
        color(channelColor) { italic { append(" $it") } }
    }
}
```